torch/csrc/jit/ir_import.cpp

```diff
@@ -0,0 +1,559 @@
+      tensor = at::CPU(at::kByte).tensor();
+      break;
+    case onnx_TensorProto_DataType_INT8:
+      tensor = at::CPU(at::kChar).tensor();
+      break;
+    case onnx_TensorProto_DataType_INT16:
+      tensor = at::CPU(at::kShort).tensor();
+      break;
+    case onnx_TensorProto_DataType_INT32:
+      tensor = at::CPU(at::kInt).tensor();
+      break;
+    case onnx_TensorProto_DataType_INT64:
+      tensor = at::CPU(at::kLong).tensor();
+      break;
+    case onnx_TensorProto_DataType_FLOAT16:
+      tensor = at::CPU(at::kHalf).tensor();
+      break;
+    case onnx_TensorProto_DataType_FLOAT:
+      tensor = at::CPU(at::kFloat).tensor();
+      break;
+    case onnx_TensorProto_DataType_DOUBLE:
+      tensor = at::CPU(at::kDouble).tensor();
+      break;
+    default:
+      throw std::runtime_error("Unsupported data type");
+  }
+
+  tensor.resize_(tensor_.dims);
+
+  TORCH_ASSERT(tensor.storage()->size() * tensor.storage()->elementSize() == tensor_.raw_data.size());
+
+  std::memcpy(tensor.data_ptr(), tensor_.raw_data.data(), tensor_.raw_data.size());
+
+  return tensor;
+}
+
+std::vector<at::Tensor> buildTensors(const std::vector<Tensor_>& tensors_) {
+  std::vector<at::Tensor> tensors;
+  for (auto t : tensors_) {
+    tensors.push_back(buildTensor(t));
+  }
+  return tensors;
+}
+
+Graph_ readSubgraph(const std::string& serialized_subgraph) {
+  pb_istream_t istream = pb_istream_from_buffer(reinterpret_cast<const pb_byte_t *>(serialized_subgraph.data()), serialized_subgraph.size());
+
+  return Reader<Graph_>::read(&istream);
+}
+
+std::vector<Graph_> readSubgraphs(const std::vector<std::string>& serialized_subgraphs) {
+  std::vector<Graph_> subgraphs;
+  for (auto serialized_subgraph : serialized_subgraphs) {
+    pb_istream_t istream = pb_istream_from_buffer(reinterpret_cast<const pb_byte_t *>(serialized_subgraph.data()), serialized_subgraph.size());
+
+    auto subgraph = Reader<Graph_>::read(&istream);
+
+    subgraphs.push_back(std::move(subgraph));
+  }
+
+  return subgraphs;
+}
+
+std::shared_ptr<Graph> buildGraph(const Graph_& graph_);
+
+std::vector<std::shared_ptr<Graph>> buildGraphs(const std::vector<Graph_>& graphs_);
+
+void buildBlocks(const std::vector<Graph_>& graphs_, Node* node,
+                 std::unordered_map<std::string, Value*>& value_map);
+
+void buildBlock(const Graph_& graph_, Block* block,
+                std::unordered_map<std::string, Value*>& value_map) {
+
+  for (auto input : graph_.inputs) {
+    value_map[input.name] = block->addInput();
+  }
+
+  for (auto node_ : graph_.nodes) {
+    TORCH_ASSERT(node_.op_type != "CppOp");
+    TORCH_ASSERT(node_.op_type != "PythonOp");
+
+    auto node = block->owningGraph()->create(Symbol::fromQualString(node_.op_type),
+                                             node_.outputs.size());
+
+    for (auto attr : node_.attrs) {
+      Symbol name = Symbol::attr(attr.name);
+
+      if (attr.type && onnx_AttributeProto_AttributeType_GRAPHS &&
```

Comment:
Why are you checking `attr.type`? Isn't that an enum? Also, you could just put an `if (attr.name == "_blocks")` as part of the `GRAPHS` case below to simplify the code.